qemu/lib/iozone.py

```diff
@@ -0,0 +1,123 @@
+    Prepare iozone env, run iozone on guest, clean up iozone env.
+    """
+    def __init__(self, test, params, vm, session):
+        """
+        run setup of iozone test
+        """
+        self.params = params
+        self.session = session
+        self.test = test
+        self.vm = vm
+        self.results_path = ""
+        self.timeout = float(params.get("iozone_timeout"))
+        check_cmd = params.get("check_cmd")
+        download_cmd = params.get("download_cmd")
+        setup_cmd = params.get("setup_cmd")
+        install_status = self.install_check(check_cmd)
+        if not install_status:
+            self.setup(download_cmd, setup_cmd)
+
+    def __session_check(self):
+        """
+        check if session is alive, if not, start a new session
+        """
+        if not self.session.is_alive():
+            self.session = self.vm.wait_for_login()
+
+    def install_check(self, check_cmd):
+        """
+        check if iozone installed in guest
+        """
+        if check_cmd:
+            output = self.session.cmd_output(check_cmd)
+            if "iozone -h" in output:
+                return True
+            else:
+                return False
+
+    def setup(self, download_cmd, setup_cmd):
+        """
+        Download iozone and install it in guest
+        """
+        if download_cmd and setup_cmd:
+            logging.info("Download iozone tarball")
+            self.session.cmd(download_cmd, timeout=self.timeout)
+            logging.info("Setup iozone")
+            self.session.cmd(setup_cmd, timeout=self.timeout)
+
+    def run(self, test_cmd, disk_letter):
+        """
+        Run iozone test in guest
+
+        :param test_cmd: iozone test cmd, like iozone -a or iozone -aZR -f
+                         I:\testiozone
+        :param disk_letter: to specify iozone_results for disks,like
+                            iozone_result_vda
+        :return: iozone run status and result.
+        """
+        # run iozone on the same device in parallel is not supported
+        self.__session_check()
+        status, output = self.session.cmd_status_output(test_cmd,
+                                                        timeout=self.timeout)
+        if self.params.get("record_results", "yes"):
+            self.results_path = os.path.join(self.test.resultsdir,
+                                             'iozone_result_%s' % disk_letter)
+            logging.debug("Write results to %s" % self.results_path)
+            try:
+                with open(self.results_path, 'a') as iozone_result:
+                    iozone_result.write(output)
+            except:
+                error_msg = "Failed to write iozone test results to file:"
+                error_msg += "%s" % self.results_path
+                os.remove(self.results_path)
+                raise exceptions.TestFail(error_msg)
+
+            if self.params.get("post_results", "no") == "yes":
+                self.post_result(disk_letter)
+
```

Comment:
I thought call session.close() here so that could execute much more iozone threads which the num of threads over max of open files(ulimit -a) , maybe it's better...